pkg/content/cache.go

```diff
@@ -0,0 +1,79 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package content
+
+import (
+	"github.com/goharbor/acceleration-service/pkg/remote"
+	lru "github.com/hashicorp/golang-lru/v2"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type RemoteCache struct {
+	// remoteCache is a LRU cache for remote image layers.The key is the string of original layer digest,
+	// and the value is the target layer descriptor after conversion.
+	remoteCache *lru.Cache[string, ocispec.Descriptor]
+	// cacheRef is the remote cache reference.
+	cacheRef string
+	// host is remote credential host.
```

Comment:
host is a func to provide registry credential by host name.

Reply:
fixed